src/routes/api/api-reset-subscription-failed-tasks.ts

```diff
@@ -0,0 +1,39 @@
+import { Request, Response } from "express";
+import { RepoSyncState } from "models/reposyncstate";
+import { Subscription } from "models/subscription";
+
+export const ApiResetSubscriptionFailedTasks = async (req: Request, res: Response): Promise<void> => {
+	const subscriptionId = req.body.subscriptionId;
+	const targetTasks = req.body.targetTasks as string[] || ["pull", "branch", "commit", "build", "deployment"];
+
+	if (!subscriptionId) {
+		res.status(400).send("please provide subscriptionId");
+		return;
+	}
+
+	const subscription = await Subscription.findByPk(subscriptionId);
+
+	if (!subscription) {
+		res.status(400).send("subscription not found");
+		return;
+	}
+
+	const log: any[] = [];
```

Comment:
Can we tighten up the typing here? Chucking in any isn't much different to not adding a type at all. Something like:
```
interface Log {
	repoSyncStateId: number,
	targetTask: "pull" | "branch" | "commit" | "build" | "deployment" | string
}
...
const log: Log[] = [];
```
should do the trick

Reply:
good point :)